### DISCUSSION TITLE
What metrics are exported in akka-actor?

### DISCUSSION THREAD

**Discussion by saffyA** [open-telemetry/opentelemetry-java-instrumentation#4909] [Q&A]
Hello,

I have enabled akka-actor & akka-http instrumentation in my application and using prometheus as exporter. However I do not see any akka actor related metrics on the exposed url (0.0.0.0:9464). Could I be missing something here?

**Comment by mateuszrzeszutek** [comment]
Hey @saffyA ,
The `akka-http` instrumentation only emits [HTTP client](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientMetrics.java) and [HTTP server](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerMetrics.java) metrics. We don't have any other metrics instrumentations for akka.

**Reply by ginaluca** [reply]
Hi @mateuszrzeszutek, I also had expectations about getting actor-related metrics. I spotted classes like `AkkaActorInstrumentationModule` in the source and that made me hope (I have no experience of the source though). 
Can I assume that the presense of a `akkaactor` package means that we'll eventually have metrics for that part as well? Would that package be a starting point if I want to try to contribute in that direction?

**Reply by mateuszrzeszutek** [reply]
Hey @ginaluca ,

> Can I assume that the presense of a `akkaactor` package means that we'll eventually have metrics for that part as well? Would that package be a starting point if I want to try to contribute in that direction?

Yes, that package/module is a good place to add akka metrics there, it's a good starting point. As (probably 😄 ) none of the maintainers of this project are experienced with scala or akka, we would gladly accept external contributions to that module.

> I spotted classes like `AkkaActorInstrumentationModule` in the source and that made me hope (I have no experience of the source though).

The `InstrumentationModule` is essentially the "main" class of any automatic javaagent instrumentation. If you want to understand the codebase better, perhaps some of our contributing docs might help you with that -- like [this one describing the general agent components](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/javaagent-jar-components.md), or [this one on instrumentation modules](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/writing-instrumentation-module.md).
